lib/dao.js

```diff
@@ -1597,6 +1598,22 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
+          self._coerce(clause, props[model].type.definition.properties)[prop];
```

Comment:
@bajtos 
To reproduce the problem please see 2a2589b04b453922f100e40ab49726742185d5d5

Not very sure if it's the right place to put the test case.

That if defined
```
"meta": {
  "type": "Object"
},
```
Then query 
```
Model.find({where: {'meta.foo': 'bar'}}) // Throws that TypeError
```

Pull request #1444 Should fix the issue. 
(Which is just applying your code suggested above and adding the test case.)

Guess the `type.definition` doesn't exists for Object type or ANY type.